R/check_element.R

```diff
@@ -1,46 +1,26 @@
-#' @export
-check_element <-
-function(curformula,elementname){
-
-        curformula<-as.character(curformula)
-        strsplit_var<-strsplit(curformula,split="")
-
-        strsplit_elem<-strsplit(elementname,split="")
-        elem_len<-length(strsplit_elem[[1]])
-
-        g3<-gregexpr(curformula,pattern=paste(elementname,"[0-9]*",sep=""))
-
-	numelement<-0
-	
-	if(length(g3)>0){
-	if(elem_len>1){
-		
-		 regexp_len3<-attr(g3[[1]],"match.length")[1]
-		if(regexp_len3>2){
-		regexp_len3<-attr(g3[[1]],"match.length")[1]-elem_len+1
-		}else{
-		regexp_len3<-attr(g3[[1]],"match.length")[1]-elem_len	
-		}
-	}else{
-		 regexp_len3<-attr(g3[[1]],"match.length")[1]-elem_len
-	}
-	
-        if(regexp_len3>=0){
-
-                if(regexp_len3==0){
-                        numelement<-1
-
-                }else{
-                         numelement<-paste(strsplit_var[[1]][(g3[[1]][1]+elem_len):(g3[[1]][1]+regexp_len3)],collapse="")
-
-                       numelement<-as.numeric(numelement)
-
-                }
-        }else{
-                numelement<-0
-
-        }
-
-	}
-        return(numelement)
-}
+#' @export
+check_element <- function(curformula, elementname) {
```

Comment:
This function is also present in a refactored state in the `main` branch: https://github.com/RECETOX/recetox-xMSannotator/blob/5725f16473c392cb9b574284953d148c334f68d0/xmsannotator/R/simple_annotation.R#L1
I'm not implying that this was unnecessary — the original code was hard to read. Just pointing it out so you don't have to spend time in case you want to refactor it further.

Reply:
Good point, but actually the refactored version and the one in main behave differently - we try to even keep known bugs, just make sure to document them along the way [here](https://github.com/RECETOX/recetox-xMSannotator/blob/main/docs/possible_issues.md)

Reply:
Yes, thanks for stressing that out! 
I just felt like small refactoring would be suitable as the code was quite messy.